src/private/MDMViewControllerTransitionCoordinator.m

```diff
@@ -18,18 +18,25 @@
 
 #import "MDMTransition.h"
 
-@interface MDMViewControllerTransitionContext : NSObject <MDMTransitionContext>
-@property(nonatomic, strong) id<UIViewControllerContextTransitioning> transitionContext;
-@property(nonatomic, strong) id<MDMTransition> transition;
+@class MDMViewControllerTransitionContextNode;
+
+@protocol MDMViewControllerTransitionContextNodeParent <NSObject>
+- (void)childNodeTransitionDidEnd:(MDMViewControllerTransitionContextNode *)childNode;
 @end
 
-@protocol MDMViewControllerTransitionContextDelegate <NSObject>
-- (void)transitionContextDidEnd:(MDMViewControllerTransitionContext *)context;
+@interface MDMViewControllerTransitionContextNode : NSObject <MDMTransitionContext, MDMViewControllerTransitionContextNodeParent>
+@property(nonatomic, strong) id<UIViewControllerContextTransitioning> transitionContext;
+@property(nonatomic, strong, readonly) id<MDMTransition> transition;
+@property(nonatomic, copy, readonly) NSMutableArray<MDMViewControllerTransitionContextNode *> *children;
 @end
 
-@implementation MDMViewControllerTransitionContext {
+@implementation MDMViewControllerTransitionContextNode {
+  // Every node points to the same array in memory.
```

Comment:
This is a minor simplification: by sharing the same mutable array, every node can add to a single shared pool of completion blocks. The alternative would be to route messages up the parent hierarchy, but given that this is a private implementation detail I wasn't particularly concerned with such an approach.